### DISCUSSION TITLE
How to configure alerts to receive notifications when a service instance goes online or offline

### DISCUSSION THREAD

**Discussion by liuhao710118** [apache/skywalking#12766] [Q&A] [answered]
How to configure alerts to receive notifications when a service instance goes online or offline?

skywalking version: 9.7.0

**Comment by wu-sheng** [comment]
You need ro determine what is offline, we don't have that.

**Reply by liuhao710118** [reply]
When the OAP server does not receive data from the agent for a long time, it is considered to be offline

Does Skywalking have heartbeat detection for service instance agents?

**Reply by wu-sheng** [reply, marked as answer]
We have the heartbeat, but we don't measure that. Because in today's env, scaling up and down is a very common case. We usually don't need that noise.

**Reply by liuhao710118** [reply]
Our company is a bit special, and now we need this kind of notification. I want to use heartbeat to monitor the online and offline of service instances. Where should I make changes to the source code

**Reply by wu-sheng** [reply]
I don't know that. This is never in the design. You can't do that in upstream alert kernel.

Maybe you could read the latest service instance in the last N minutes and compare to your known list?

**Reply by wu-sheng** [reply]
In most cases, this alerting belongs to your deployment platform, such as k8s has this status.

**Reply by liuhao710118** [reply]
Okay, thank you for your real-time response
